pkg/sql/copy_from.go

```diff
@@ -692,6 +698,17 @@ func (c *copyMachine) currentBatchSize() int {
 }
 
 func (c *copyMachine) readTextData(ctx context.Context, final bool) (brk bool, err error) {
+	// If we only expect a single row, we avoid unnecessary copies
+	// and quadratic search behaviour by assuming we won't find
+	// the delimiter.
+	if c.singleRowExpected {
```

Comment:
i wonder if it might make sense to avoid the unnecessary copies by switching to a []byte. i tried it out here, since it might help the other `read*Data` functions too: https://github.com/cockroachdb/cockroach/pull/106088

Reply:
I'll pull that commit into this branch with a few fixes after lunch.

Reply:
Looks like I've still got one a problem causing the docker tests to fail. I'll take another look this evening or tomorrow morning.